code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm

```diff
@@ -98,4 +95,159 @@
+	temp_turfs += infront
+	if (!(!infront || infront.density))
+		temp_turfs += infront_left
+	if (!(!infront || infront.density))
+		temp_turfs += infront_right
+
+	for (var/turf/current_turfs in temp_turfs)
+
+		if (!istype(current_turfs))
+			continue
+
+		if (current_turfs.density)
+			continue
+
+		target_turfs += current_turfs
+		telegraph_atom_list += new /obj/effect/xenomorph/xeno_telegraph/red(current_turfs, 2)
+
+	for (var/turf/current_turfs in target_turfs)
+		for (var/mob/living/carbon/target in current_turfs)
+			if (target.stat == DEAD)
+				continue
+
+			if (!isXenoOrHuman(target) || xeno.can_not_harm(target))
+				continue
+
+			xeno.visible_message(SPAN_DANGER("[xeno] scratches [target] all around its body!"), \
+			SPAN_XENOWARNING("You slash [target] multiple times!"))
+			xeno.flick_attack_overlay(target, "slash")
+			target.last_damage_data = create_cause_data(xeno.caste_type, xeno)
+			log_attack("[key_name(xeno)] attacked [key_name(target)] with Flurry")
+			target.apply_armoured_damage(30, ARMOR_MELEE, BRUTE)
+			playsound(get_turf(target), 'sound/weapons/alien_claw_flesh4.ogg', 30, TRUE)
+			xeno.emote("roar")
+			xeno.flick_heal_overlay(1 SECONDS, "#00B800")
+			xeno.gain_health(30)
+			..()
+
+/datum/action/xeno_action/activable/tail_jab/use_ability(atom/targeted_atom)
+	var/mob/living/carbon/Xenomorph/xeno = owner
+	var/mob/living/carbon/target = targeted_atom
+	var/distance = get_dist(xeno, target)
+
+	if(!action_cooldown_check())
+		return
+
+	if(!xeno.check_state())
+		return
+
+	var/list/turf/path = getline2(xeno, target, include_from_atom = FALSE)
+	for(var/turf/path_turf as anything in path)
+		if(path_turf.density)
+			to_chat(xeno, SPAN_WARNING("There's something blocking you from stricking!"))
+			return
+		var/atom/barrier = path_turf.handle_barriers(A = xeno , pass_flags = (PASS_MOB_THRU_XENO|PASS_OVER_THROW_MOB|PASS_TYPE_CRAWLER))
+		if(barrier != path_turf)
+			to_chat(xeno, SPAN_WARNING("There's something blocking you from stricking!"))
+			return
+		for(var/obj/structure/current_structure in path_turf)
+			if(istype(current_structure, /obj/structure/window/framed))
+				var/obj/structure/window/framed/target_window = current_structure
+				if(target_window.unslashable)
+					return
+				target_window.shatter_window(TRUE)
+				apply_cooldown(cooldown_modifier = 0.5)
+				xeno.visible_message(SPAN_XENOWARNING("\The [xeno] strikes the window with their tail!"), SPAN_XENOWARNING("You strike the window with your tail!"))
+				playsound(get_turf(target_window),'sound/effects/glassbreak3.ogg', 30, TRUE)
+				return
+
+	if(!isXenoOrHuman(target) || xeno.can_not_harm(target))
+		xeno.visible_message(SPAN_XENOWARNING("\The [xeno] swipes their tail through the air!"), SPAN_XENOWARNING("You swipe your tail through the air!"))
+		apply_cooldown(cooldown_modifier = 0.1)
+		playsound(xeno, 'sound/effects/alien_tail_swipe1.ogg', 50, TRUE)
+		return
+
+	if(distance > 2)
+		return
```

Comment:
above user is correct move this early return diego...

Reply:
fixed . i will upload the fix soon